packages/sync/src/class-functions.php

```diff
@@ -630,19 +630,30 @@ public static function get_paused_plugins() {
 	 * Return the theme's supported features.
 	 * Used for syncing the supported feature that we care about.
 	 *
+	 * @param WP_Theme $theme Theme Object.
+	 *
 	 * @return array List of features that the theme supports.
 	 */
-	public static function get_theme_support() {
+	public static function get_theme_support( $theme = null ) {
 		global $_wp_theme_features;
 
 		$theme_support = array();
+
+		if ( ! $theme instanceof WP_Theme ) {
+			$theme = wp_get_theme();
+		}
 		foreach ( Defaults::$default_theme_support_whitelist as $theme_feature ) {
```

Comment:
In Sync we get_theme_support for both the current theme and the previous during switch_theme actions.
Before getting theme features we should check if $theme is the current theme, if not don't add the support info otherwise the previous theme data will contain supported features for the current theme.

Reply:
actually we can ignore this as previous_theme passes the false flag so it doesn't actually call this function, however it still applies that if theme passed isn't current we shouldn't populate.

Reply:
@enejb I was testing this and I do think we should be explicit. As the foreach references the current theme we should do a check so $theme_support only populates if the theme is the current theme.

Or we leave ourselves open to future issues if $previous_theme gets used on WP.com side.